main/lsp/LSPTypechecker.cc

```diff
@@ -214,7 +215,13 @@ vector<core::FileRef> LSPTypechecker::runFastPath(LSPFileUpdates &updates, Worke
         auto fref = gs->findFileByPath(file->path());
         ENFORCE(fref.exists(), "New files are not supported in the fast path");
 
-        if (file->isPackage(*gs)) {
+        // We update the file even if it's a `__package.rb` file, as this will ensure that we propagate the
+        // `isOpenInClient` flag. Additionally, the indexer will only select the fast path for a `__package.rb` if the
+        // file's contents match what's already in the file table, so this will be safe.
+        auto oldFile = gs->replaceFile(fref, std::move(file));
+
+        if (oldFile->isPackage(*gs)) {
+            packageFiles.emplace_back(fref);
```

Comment:
Alternatively: is there harm to always computing the oldFoundHashesForFiles for `__package.rb` files?

Soon we're going to need to do this anyways, so that we can have `__package.rb` file edits actually take the fast path.

If there's no harm to computing and ignoring them, it would be nice to start that now.

Reply:
I don't think there's any harm in doing this. I'll experiment with handling package files the same as source files.